### PR TITLE
Radiation pulse optimization

### PR DESCRIPTION
**Pull request by ghost**
The pulse will now do the calculation of (severity \* 0.5) before the proc's effects and put it into a variable, light_severity. This is used in place of the old (severity / 2) calculation called on _every atom_ affected by the pulse. Also disables radiation pulses from a gravity generator being shown in asay, as requested by KorPhaeron.

### CODE REVIEW

File: code/game/objects/radiation.dm [hunk 1]
```diff
@@ -11,6 +11,7 @@
 	if(heavy_range > light_range)
 		light_range = heavy_range
 
+	var/light_severity = severity / 2
```
Comment (RemieRichards) [inline code/game/objects/radiation.dm]: No, ```severity *0.5```

Comment (Xhuis) [inline code/game/objects/radiation.dm]: Mmkay vOv

**Comment by duncathan** [comment]
They don't do the same thing; * 0.5 is faster.

**Comment by RemieRichards** [comment]
what @duncathan said, learn about Arithmetic Logic Units nerd.

**Comment by KorPhaeron** [comment]
I dunno if there is any easy fix to this or if it's worth an issue report but the grav gen was spamming pulse notifications in asay the other day

**Comment by Xhuis** [comment]
@KorPhaeron That is an easy fix, I'll disable it.

**Comment by Xhuis** [comment]
The gravity generator's rad pulses won't be logged or spammed in asay anymore.

**Comment by KorPhaeron** [comment]
Thanks

**Comment by Xhuis** [comment]
This is a pretty high-priority merge as it will reduce the load on the server by a fairly staggering amount. Please merge this when you can.

**Comment by Razharas** [comment]
I thought most languages approximate devision with multiplication anyway
But fine

**Pull request merged**